homeassistant/components/switchbee/switch.py

```diff
@@ -25,37 +25,36 @@ async def async_setup_entry(
 ) -> None:
     """Set up Switchbee switch."""
     coordinator: SwitchBeeCoordinator = hass.data[DOMAIN][entry.entry_id]
-    device_types = (
-        [DeviceType.TimedPowerSwitch]
-        if coordinator.switch_as_light
-        else [
+
+    async_add_entities(
+        Device(hass, device, coordinator)
+        for device in coordinator.data.values()
+        if device.type
+        in [
             DeviceType.TimedPowerSwitch,
             DeviceType.GroupSwitch,
             DeviceType.Switch,
             DeviceType.TimedSwitch,
-            DeviceType.TwoWay,
         ]
     )
 
-    async_add_entities(
-        Device(hass, device, coordinator)
-        for device in coordinator.data.values()
-        if device.type in device_types
-    )
-
 
-class Device(CoordinatorEntity, SwitchEntity):
+class Device(CoordinatorEntity[SwitchBeeCoordinator], SwitchEntity):
```

Comment:
This is not a device. It's an entity. Please rename to something ending with `Entity`.